eng/Analyzers.props

```diff
@@ -0,0 +1,16 @@
+<Project>
+<!-- StyleCop.Analyzers configuration-->
+<!-- If the runtime is unknown, GeneratePlatformNotSupportedAssembly is set in 
+System.Device.Gpio.csproj, which creates an auto-generated code 
+file which is not marked as such, making StyleCop fail. -->
+  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true' AND '$(MSBuildProjectFile)'!='Iot.Device.Bindings.csproj'">
+    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)\StyleCop.Analyzers.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+  <ItemGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true' AND '$(MSBuildProjectFile)'!='Iot.Device.Bindings.csproj'">
```

Comment:
If this item is specific to Iot.Device.Bindings, then I would suggest to put it only on that project not on a general props file. Also, if it is applicable to that project, we will want to instead also make it applicable to all projects under src/devices so that developers working on individual projects will also get this feedback during development.

Reply:
The `AND '$(MSBuildProjectFile)'!='Iot.Device.Bindings.csproj'` part is temporary only. Without this, the build would fail for now, as that csproj file directly imports (root)/Directory.Build.props, which has the rules already enabled. This will be removed once all parts are merged.

Reply:
Yup I missed that, this sounds good to me.